binary_transparency/firmware/internal/verify/bundle.go

```diff
@@ -57,6 +57,39 @@ func BundleForUpdate(bundleRaw, fwHash []byte, dc api.LogCheckpoint, cpFunc Cons
 	return nil
 }
 
+// BundleValidateWitness verifies the received bundle against witness checkpoint
+func BundleValidateWitness(bundleRaw []byte, wc api.LogCheckpoint, cpFunc ConsistencyProofFunc) error {
+	var pb api.ProofBundle
+	lv := NewLogVerifier()
+	if err := json.Unmarshal(bundleRaw, &pb); err != nil {
+		return fmt.Errorf("failed to parse proof bundle: %w", err)
+	}
```

Comment:
I'd be tempted to return the parsed ProofBundle from the method above, and then pass it back into this method if needed. Parsing it twice feels like wasted work, and just a little uncomfortable if the two parse methods ever diverged.

Reply:
Agree, I have made the desired changes in the new revision, kindly check!